[TM] ADD: Меняем работу кобуры на парадайз-стайл

Mirag1993 opened the pull request:
## Описание / Что этот PR делает
Изменяет работу кобуры , теперь это не только дополнительный карман на рубашке а удобный предмет для быстрой работы с оружием.

## Причина создания ПР / Почему это хорошо для игры
Мне всегда не нравилась механика кобур ТГ , открой доп карман , кликни мимо кобуры , умри пока доставал пушку , а если еще чтото в руке есть , то надо сначало сбросить , что не всегда из за лагов удается быстро сделать.
Стиль парадиз кобур вводит четкое обещание , после нажатия кнопки , то что было в кобуре , появится в активной руке.

## Демонстрация изменений / Тестирование
1.Установить кейбинд на удобную кнопку
2.Надеть кобуру на одежду либо взять в руки( не сработает кейбинд , только в ручную работать )
3.Взять оружие подходящее для кобуры , нажать кейбинд либо кликнуть оружием по кобуре или одежде с кобурой
4.Оружие вложится в кобуру если позволяют условия
5.Достать можно либо повторным нажатием на кейбинд либо альт клик по униформе с кобурой либо кобуре , откроется ui окно с содержимым
6.Оружие снова появится в руке.

## Список изменений
Изменена работа кобуры , теперь она имеет всего один слот.
В кобуру можно положить только оружие смолл размера и без авто режима.
Кобура nuke может принимать оружие большего размера.
Добавлена unbind кнопка для действия holster/unholster.


```yml
🆑
- add: Новая система кобур в стиле Paradise
- add: Поддержка клавиши для быстрого доступа к кобуре
- add: Проверка размера оружия (TINY/SMALL)
- add: Максимум 1 предмет в кобуре 
- add: Возможность положить в кобуру пистолет, револьвер и энергетическое оружие
- del: Старая система кобур Shiptest (закомментирована)
/🆑
```

Mirag1993 committed:
EDIT: Меняем работу кобуры на парадайз-стайл

Меняем работу кобуры на парадайз-стайл

KOCMODECAHTHUK commented on mod_celadon/qol/holster_paradise/code/holster_types.dm:
```diff
@@ -0,0 +1,205 @@
+
+/obj/item/clothing/accessory/holster/chameleon/broken/Initialize()
+	. = ..()
+	chameleon_action.emp_randomise(INFINITY)
```
А зачем...?

Mirag1993 replied:
зачем что ? зачем кобура хамелеон ? ну хз , сложный вопрос

KOCMODECAHTHUK reviewed (changes requested):
Это конечно хорошо, но почему не улучшить тогда обычный холстер без переписывания методов?

Мне не очень нравится вообще что у нас в билде существуют функциональные предметы что цепляются на одежду, это только путает.

Если хочешь улучшить попробуй через метод `on_alt_click_async(datum/source, mob/user)` что я разбирал в #2215
Ну и хоткей добавить отдельно с той же проверкой на холстер и оружие.

Mirag1993 committed:
Фикс 2

Реворк чутка , добавлена возможность смотреть инвентарь кобуры , изменена проверка

Mirag1993 committed:
Фикс 3

Смешные баги с бесконечным хранилищем пистолетов

KOCMODECAHTHUK commented on code/controllers/subsystem/input.dm:
```diff
@@ -82,7 +82,7 @@ VERB_MANAGER_SUBSYSTEM_DEF(input)
 		"F" = "F", // intent left
 		"G" = "G", // intent right
-		"H" = "H", // stop pulling
+		"C" = "C", // stop pulling
```
Конфликт с пиксель-шифтингом

KOCMODECAHTHUK commented on mod_celadon/qol/holster_paradise/code/holster_types.dm:
```diff
@@ -0,0 +1,339 @@
+	return ..()
+
+/obj/item/clothing/accessory/holster/proc/can_holster(obj/item/I)
+	if(!istype(I, holster_allow))
+		return FALSE
+
+	// Для кобуры нюкера разрешаем крупное оружие (может держать почти любое баллистическое оружие)
+	if(istype(src, /obj/item/clothing/accessory/holster/nukie))
+		return TRUE
+
+	var/obj/item/gun/G = I
+	if(istype(G))
```
А зачем?
У тебя уже же проверка ранее есть
```
if(!istype(I, holster_allow))
		return FALSE
```

KOCMODECAHTHUK commented on mod_celadon/qol/holster_paradise/code/holster_types.dm:
```diff
@@ -0,0 +1,339 @@
+		// БЛОК: среднее и тяжелое оружие (SMG, винтовки)
+		// Если в базе ваш USP имеет MEDIUM — либо ослабьте правило, либо добавьте исключение ниже.
+		if(!(istype(G, /obj/item/gun/ballistic/automatic/pistol) || istype(G, /obj/item/gun/ballistic/revolver)))
+			if(G.weapon_weight != WEAPON_LIGHT)
+				return FALSE
```
То же самое, у тебя ранее проверка есть или я чет не понимаю?
```
		if(G.w_class > WEIGHT_CLASS_NORMAL)
			return FALSE
```

KOCMODECAHTHUK commented on mod_celadon/qol/holster_paradise/code/holster_types.dm:
```diff
@@ -0,0 +1,339 @@
+	// Используем проверку can_holster() из кобуры
+	if(istype(original_parent, /obj/item/clothing/accessory/holster))
+		var/obj/item/clothing/accessory/holster/holster = original_parent
+		if(!holster.can_holster(I, M))
```
Ты уверен что тут нет ошибки?
У тебя везде функция принимает всего 1 аргумент, а тут 2 аргумента.

Mirag1993 replied:
Спасибо , это была заглушка за которую я забыл

KOCMODECAHTHUK reviewed (changes requested):
Тяжело, сильно много багов будет...
Это я ещё про дубляж кода умолчал.

Mirag1993 committed:
Merge branch 'CeladonSS13:beta-dev' into Кобура-(Paradise)

Mirag1993 committed:
Предфинальный фикс

Попытка рефактора , надеюсь удачная , чуть разнесли код и убрали косяки

Mirag1993 committed:
Финальный штрих

Mirag1993 committed:
Финал-Финал фикс

Убираем лишнюю проверку

Mirag1993 committed:
Фикс 6

Исправляем претензии по коду

Mirag1993 committed:
Финал финал финал

Надеюсь последние правки , косметику править уже не буду

KOCMODECAHTHUK commented on mod_celadon/qol/holster_paradise/code/holster_logging.dm:
Напомни а зачем нам такое подробное логирование кобуры?

KOCMODECAHTHUK commented on mod_celadon/qol/holster_paradise/code/holster_types.dm:
Выглядит так как будто это лишнее

KOCMODECAHTHUK reviewed (changes requested):
Чет смущает подробнейшая логирование

Mirag1993 committed:
Убираем лишние логирование

Убираем лишние логирование

MysticalFaceLesS commented:
1. Дописать модульность.
2. Убрать логирование.
3. Сделать правильную иницивлизацию файлов.

Mirag1993 committed:
Merge branch 'CeladonSS13:beta-dev' into Кобура-(Paradise)

Mirag1993 committed:
Update

Update

Mirag1993 commented:
ну впринципе если никому не нужно то ладно

Mirag1993 committed:
Delete debug

Удаляю дебаг , ревьювер даже с помощью ридми не может читать код , его тоже убираю

Mirag1993 committed:
Delete comments, kiss

Ревьювер не смог прочитать код ни с ридми ни с комментариями , значит они не нужны

Mirag1993 committed:
Final
